sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -248,6 +248,26 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
   }
 
   /**
+   * Create a partition range specification map.
+   * Note, we do not support partition range yet. Thus, only support the partition spec specified
+   * in the following format: partition_column=val
+   */
+  override def visitPartitionRangeSpec(
+      ctx: PartitionRangeSpecContext): Map[String, String] = withOrigin(ctx) {
+    ctx.partitionRangeVal.asScala.map { pVal =>
+      val operator = pVal.comparisonOperator().getChild(0).asInstanceOf[TerminalNode]
+      if (operator.getSymbol.getType != SqlBaseParser.EQ) {
```

Comment:
I will revert it. The purpose here is to issue a better exception message if users specify predicates in alter table drop partition.